### ISSUE TITLE
Update failure from 2.6.9 to 2.7.3

### ISSUE THREAD

**Issue by dandunckelman** [rundeck/rundeck#2378]
Issue type: Bug report

**My Rundeck detail**

* Rundeck version: 2.6.9
* install type: apt
* OS Name/version: Ubuntu 16.04

**Expected Behavior**
Updating from 2.6.9 to 2.7.3 works as expected.

**Actual Behavior**
I get this error in `/var/log/rundeck/service.log`:

```
2017-03-20 21:38:30,314 [main] ERROR hbm2ddl.SchemaUpdate  - HHH000388: Unsuccessful: alter table base_report add column failed_node_list longvarchar                                                                                 
2017-03-20 21:38:30,315 [main] ERROR hbm2ddl.SchemaUpdate  - General error: "java.lang.IllegalArgumentException: timeNanos out of range 86400754000000"; SQL statement:                                                               
alter table base_report add column failed_node_list longvarchar [50000-193]                                                                                                                                                           
2017-03-20 21:38:31,620 [main] ERROR hbm2ddl.SchemaUpdate  - HHH000388: Unsuccessful: alter table base_report add column filter_applied longvarchar                                                                                   
2017-03-20 21:38:31,620 [main] ERROR hbm2ddl.SchemaUpdate  - General error: "java.lang.IllegalArgumentException: timeNanos out of range 86400754000000"; SQL statement:                                                               
alter table base_report add column filter_applied longvarchar [50000-193]                                                                                                                                                             
2017-03-20 21:38:32,745 [main] ERROR hbm2ddl.SchemaUpdate  - HHH000388: Unsuccessful: alter table base_report add column succeeded_node_list longvarchar                                                                              
2017-03-20 21:38:32,746 [main] ERROR hbm2ddl.SchemaUpdate  - General error: "java.lang.IllegalArgumentException: timeNanos out of range 86400754000000"; SQL statement:                                                               
alter table base_report add column succeeded_node_list longvarchar [50000-193]                                                                                                                                                        
2017-03-20 21:38:33,235 [main] ERROR hbm2ddl.SchemaUpdate  - HHH000388: Unsuccessful: alter table execution add column execution_type varchar(30)                                                                                     
2017-03-20 21:38:33,240 [main] ERROR hbm2ddl.SchemaUpdate  - General error: "java.lang.IllegalArgumentException: timeNanos out of range 86400754000000"; SQL statement:                                                               
alter table execution add column execution_type varchar(30) [50000-193]                                                                                                                                                               
2017-03-20 21:38:33,535 [main] ERROR hbm2ddl.SchemaUpdate  - HHH000388: Unsuccessful: alter table execution add column node_filter_editable boolean                                                                                   
2017-03-20 21:38:33,535 [main] ERROR hbm2ddl.SchemaUpdate  - General error: "java.lang.IllegalArgumentException: timeNanos out of range 86400754000000"; SQL statement:                                                               
alter table execution add column node_filter_editable boolean [50000-193]                                                                                                                                                             
INFO  BootStrap: Starting Rundeck 2.7.3-1 (2017-03-11) ...                                                                                                                                                                            
INFO  BootStrap: using rdeck.base config property: /var/lib/rundeck                                                                                                                                                                   
INFO  BootStrap: loaded configuration: /etc/rundeck/framework.properties                                                                                                                                                              
INFO  BootStrap: RSS feeds disabled                                                                                                                                                                                                   
INFO  BootStrap: Preauthentication is disabled                                                                                                                                                                                        
INFO  BootStrap: Rundeck is ACTIVE: executions can be run.                                                                                                                                                                            
WARN  BootStrap: The JVM default encoding is not UTF-8: ISO-8859-1, you may not see output as expected for multibyte locales. Specify -Dfile.encoding=UTF-8 in the JVM options.                                                       
ERROR SqlExceptionHelper: General error: "java.lang.IllegalArgumentException: timeNanos out of range 86400754000000"; SQL statement:                                                                                                  
select this_.id as y0_ from execution this_ where this_.date_started is not null and this_.date_completed is null and this_.server_nodeuuid is null and this_.date_started<? and (this_.status is null or this_.status<>?) [50000-193]
ERROR GrailsContextLoaderListener: Error initializing the application: could not extract ResultSet                                                                                                                                    
org.hibernate.exception.GenericJDBCException: could not extract ResultSet                                                                                                                                                             
        at org.grails.datastore.gorm.GormStaticApi$_withCriteria_closure11.doCall(GormStaticApi.groovy:305)                                                                                                                           
        at org.grails.datastore.mapping.core.DatastoreUtils.execute(DatastoreUtils.java:302)                                                                                                                                          
        at org.grails.datastore.gorm.AbstractDatastoreApi.execute(AbstractDatastoreApi.groovy:37)                                                                                                                                     
        at org.grails.datastore.gorm.GormStaticApi.withCriteria(GormStaticApi.groovy:304)                                                                                                                                             
        at rundeck.services.ExecutionService.cleanupRunningJobsAsync(ExecutionService.groovy:652)                                                                                                                                     
        at rundeck.services.ExecutionService.cleanupRunningJobsAsync(ExecutionService.groovy:651)                                                                                                                                     
        at BootStrap$_closure1.doCall(BootStrap.groovy:377)                                                                                                                                                                           
        at grails.util.Environment.evaluateEnvironmentSpecificBlock(Environment.java:327)                                                                                                                                             
        at grails.util.Environment.executeForEnvironment(Environment.java:320)                                                                                                                                                        
        at grails.util.Environment.executeForCurrentEnvironment(Environment.java:296)                                                                                                                                                 
        at org.eclipse.jetty.server.handler.ContextHandler.callContextInitialized(ContextHandler.java:788)                                                                                                                            
        at org.eclipse.jetty.servlet.ServletContextHandler.callContextInitialized(ServletContextHandler.java:434)                                                                                                                     
        at org.eclipse.jetty.server.handler.ContextHandler.startContext(ContextHandler.java:780)                                                                                                                                      
        at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:284)                                                                                                                               
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1322)                                                                                                                                               
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:732)                                                                                                                                           
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:490)                                                                                                                                                     
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)                                                                                                                                        
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:118)                                                                                                                                     
        at org.eclipse.jetty.server.Server.start(Server.java:342)                                                                                                                                                                     
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:100)                                                                                                                                   
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:60)                                                                                                                                          
        at org.eclipse.jetty.server.Server.doStart(Server.java:290)                                                                                                                                                                   
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)                                                                                                                                        
        at com.dtolabs.rundeck.RunServer.run(RunServer.java:119)                                                                                                                                                                      
        at com.dtolabs.rundeck.RunServer.main(RunServer.java:79)                                                                                                                                                                      
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalArgumentException: timeNanos out of range 86400754000000"; SQL statement:                                                                                   
select this_.id as y0_ from execution this_ where this_.date_started is not null and this_.date_completed is null and this_.server_nodeuuid is null and this_.date_started<? and (this_.status is null or this_.status<>?) [50000-193]
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)                                                                                                                                                       
        at org.h2.message.DbException.get(DbException.java:168)                                                                                                                                                                       
        at org.h2.message.DbException.convert(DbException.java:295)                                                                                                                                                                   
        at org.h2.command.Command.executeQuery(Command.java:213)                                                                                                                                                                      
        at org.h2.jdbc.JdbcPreparedStatement.executeQuery(JdbcPreparedStatement.java:110)                                                                                                                                             
        ... 26 more                                                                                                                                                                                                                   
Caused by: java.lang.IllegalArgumentException: timeNanos out of range 86400754000000                                                                                                                                                  
        at org.h2.value.ValueTimestamp.<init>(ValueTimestamp.java:54)                                                                                                                                                                 
        at org.h2.value.ValueTimestamp.fromDateValueAndNanos(ValueTimestamp.java:68)                                                                                                                                                  
        at org.h2.value.ValueTimestamp.fromMillisNanos(ValueTimestamp.java:95)                                                                                                                                                        
        at org.h2.store.Data.readValue(Data.java:779)                                                                                                                                                                                 
        at org.h2.index.PageDataLeaf.readRow(PageDataLeaf.java:624)                                                                                                                                                                   
        at org.h2.index.PageDataLeaf.getRowAt(PageDataLeaf.java:334)                                                                                                                                                                  
        at org.h2.index.PageDataCursor.nextRow(PageDataCursor.java:101)                                                                                                                                                               
        at org.h2.index.PageDataCursor.next(PageDataCursor.java:67)                                                                                                                                                                   
        at org.h2.index.IndexCursor.next(IndexCursor.java:305)                                                                                                                                                                        
        at org.h2.table.TableFilter.next(TableFilter.java:498)                                                                                                                                                                        
        at org.h2.command.dml.Select.queryFlat(Select.java:542)                                                                                                                                                                       
        at org.h2.command.dml.Select.queryWithoutCache(Select.java:655)                                                                                                                                                               
        at org.h2.command.dml.Query.query(Query.java:341)                                                                                                                                                                             
        at org.h2.command.dml.Query.query(Query.java:309)                                                                                                                                                                             
        at org.h2.command.dml.Query.query(Query.java:36)                                                                                                                                                                              
        at org.h2.command.CommandContainer.query(CommandContainer.java:110)                                                                                                                                                           
        at org.h2.command.Command.executeQuery(Command.java:201)                                                                                                                                                                      
        ... 27 more                                                                                                                                                                                                                   
ERROR GrailsContextLoaderListener: Error initializing Grails: could not extract ResultSet                                                                                                                                             
org.hibernate.exception.GenericJDBCException: could not extract ResultSet                                                                                                                                                             
        at org.grails.datastore.gorm.GormStaticApi$_withCriteria_closure11.doCall(GormStaticApi.groovy:305)                                                                                                                           
        at org.grails.datastore.mapping.core.DatastoreUtils.execute(DatastoreUtils.java:302)                                                                                                                                          
        at org.grails.datastore.gorm.AbstractDatastoreApi.execute(AbstractDatastoreApi.groovy:37)                                                                                                                                     
        at org.grails.datastore.gorm.GormStaticApi.withCriteria(GormStaticApi.groovy:304)                                                                                                                                             
        at rundeck.services.ExecutionService.cleanupRunningJobsAsync(ExecutionService.groovy:652)                                                                                                                                     
        at rundeck.services.ExecutionService.cleanupRunningJobsAsync(ExecutionService.groovy:651)                                                                                                                                     
        at BootStrap$_closure1.doCall(BootStrap.groovy:377)                                                                                                                                                                           
        at grails.util.Environment.evaluateEnvironmentSpecificBlock(Environment.java:327)                                                                                                                                             
        at grails.util.Environment.evaluateEnvironmentSpecificBlock(Environment.java:327)                                                                                                                                             
        at grails.util.Environment.executeForEnvironment(Environment.java:320)                                                                                                                                                        
        at grails.util.Environment.executeForCurrentEnvironment(Environment.java:296)                                                                                                                                                 
        at org.eclipse.jetty.server.handler.ContextHandler.callContextInitialized(ContextHandler.java:788)                                                                                                                            
        at org.eclipse.jetty.servlet.ServletContextHandler.callContextInitialized(ServletContextHandler.java:434)                                                                                                                     
        at org.eclipse.jetty.server.handler.ContextHandler.startContext(ContextHandler.java:780)                                                                                                                                      
        at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:284)                                                                                                                               
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1322)                                                                                                                                               
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:732)                                                                                                                                           
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:490)                                                                                                                                                     
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)                                                                                                                                        
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:118)                                                                                                                                     
        at org.eclipse.jetty.server.Server.start(Server.java:342)                                                                                                                                                                     
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:100)                                                                                                                                   
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:60)                                                                                                                                          
        at org.eclipse.jetty.server.Server.doStart(Server.java:290)                                                                                                                                                                   
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)                                                                                                                                        
        at com.dtolabs.rundeck.RunServer.run(RunServer.java:119)                                                                                                                                                                      
        at com.dtolabs.rundeck.RunServer.main(RunServer.java:79)                                                                                                                                                                      
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalArgumentException: timeNanos out of range 86400754000000"; SQL statement:                                                                                   
select this_.id as y0_ from execution this_ where this_.date_started is not null and this_.date_completed is null and this_.server_nodeuuid is null and this_.date_started<? and (this_.status is null or this_.status<>?) [50000-193]
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)                                                                                                                                                       
        at org.h2.message.DbException.get(DbException.java:168)                                                                                                                                                                       
        at org.h2.message.DbException.convert(DbException.java:295)                                                                                                                                                                   
        at org.h2.command.Command.executeQuery(Command.java:213)                                                                                                                                                                      
        at org.h2.jdbc.JdbcPreparedStatement.executeQuery(JdbcPreparedStatement.java:110)                                                                                                                                             
        ... 26 more                                                                                                                                                                                                                   
Caused by: java.lang.IllegalArgumentException: timeNanos out of range 86400754000000                                                                                                                                                  
        at org.h2.value.ValueTimestamp.<init>(ValueTimestamp.java:54)                                                                                                                                                                 
        at org.h2.value.ValueTimestamp.fromDateValueAndNanos(ValueTimestamp.java:68)                                                                                                                                                  
        at org.h2.value.ValueTimestamp.fromMillisNanos(ValueTimestamp.java:95)                                                                                                                                                        
        at org.h2.store.Data.readValue(Data.java:779)                                                                                                                                                                                 
        at org.h2.index.PageDataLeaf.readRow(PageDataLeaf.java:624)                                                                                                                                                                   
        at org.h2.index.PageDataLeaf.getRowAt(PageDataLeaf.java:334)                                                                                                                                                                  
        at org.h2.index.PageDataCursor.nextRow(PageDataCursor.java:101)                                                                                                                                                               
        at org.h2.index.PageDataCursor.next(PageDataCursor.java:67)                                                                                                                                                                   
        at org.h2.index.IndexCursor.next(IndexCursor.java:305)                                                                                                                                                                        
        at org.h2.table.TableFilter.next(TableFilter.java:498)                                                                                                                                                                        
        at org.h2.command.dml.Select.queryFlat(Select.java:542)                                                                                                                                                                       
        at org.h2.command.dml.Select.queryWithoutCache(Select.java:655)                                                                                                                                                               
        at org.h2.command.dml.Query.query(Query.java:341)                                                                                                                                                                             
        at org.h2.command.dml.Query.query(Query.java:309)                                                                                                                                                                             
        at org.h2.command.dml.Query.query(Query.java:36)                                                                                                                                                                              
        at org.h2.command.CommandContainer.query(CommandContainer.java:110)                                                                                                                                                           
        at org.h2.command.Command.executeQuery(Command.java:201)                                                                                                                                                                      
        ... 27 more                                                                                                                                                                                                                   
2017-03-20 21:38:53.229:WARN:oejw.WebAppContext:main: Failed startup of context o.e.j.w.WebAppContext@67e25252{/,file:/var/lib/rundeck/exp/webapp/,STARTING}{/var/lib/rundeck/exp/webapp}                                             
org.springframework.beans.factory.access.BootstrapException: Error executing bootstraps; nested exception is org.hibernate.exception.GenericJDBCException: could not extract ResultSet                                                
        at org.codehaus.groovy.grails.web.context.GrailsContextLoaderListener.initWebApplicationContext(GrailsContextLoaderListener.java:126)                                                                                         
        at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:106)                                                                                                                   
        at org.eclipse.jetty.server.handler.ContextHandler.callContextInitialized(ContextHandler.java:788)                                                                                                                            
        at org.eclipse.jetty.servlet.ServletContextHandler.callContextInitialized(ServletContextHandler.java:434)                                                                                                                     
        at org.eclipse.jetty.server.handler.ContextHandler.startContext(ContextHandler.java:780)                                                                                                                                      
        at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:284)                                                                                                                               
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1322)                                                                                                                                               
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:732)                                                                                                                                           
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:490)                                                                                                                                                     
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)                                                                                                                                        
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:118)                                                                                                                                     
        at org.eclipse.jetty.server.Server.start(Server.java:342)                                                                                                                                                                     
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:100)                                                                                                                                   
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:60)                                                                                                                                          
        at org.eclipse.jetty.server.Server.doStart(Server.java:290)                                                                                                                                                                   
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)                                                                                                                                        
        at com.dtolabs.rundeck.RunServer.run(RunServer.java:119)                                                                                                                                                                      
        at com.dtolabs.rundeck.RunServer.main(RunServer.java:79)                                                                                                                                                                      
Caused by:                                                                                                                                                                                                                            
org.hibernate.exception.GenericJDBCException: could not extract ResultSet                                                                                                                                                             
        at org.grails.datastore.gorm.GormStaticApi$_withCriteria_closure11.doCall(GormStaticApi.groovy:305)                                                                                                                           
        at org.grails.datastore.mapping.core.DatastoreUtils.execute(DatastoreUtils.java:302)                                                                                                                                          
        at org.grails.datastore.gorm.AbstractDatastoreApi.execute(AbstractDatastoreApi.groovy:37)                                                                                                                                     
        at org.grails.datastore.gorm.GormStaticApi.withCriteria(GormStaticApi.groovy:304)                                                                                                                                             
        at rundeck.services.ExecutionService.cleanupRunningJobsAsync(ExecutionService.groovy:652)                                                                                                                                     
        at rundeck.services.ExecutionService.cleanupRunningJobsAsync(ExecutionService.groovy:651)                                                                                                                                     
        at BootStrap$_closure1.doCall(BootStrap.groovy:377)                                                                                                                                                                           
        at grails.util.Environment.evaluateEnvironmentSpecificBlock(Environment.java:327)                                                                                                                                             
        at grails.util.Environment.executeForEnvironment(Environment.java:320)                                                                                                                                                        
        at grails.util.Environment.executeForCurrentEnvironment(Environment.java:296)                                                                                                                                                 
        at org.eclipse.jetty.server.handler.ContextHandler.callContextInitialized(ContextHandler.java:788)                                                                                                                            
        at org.eclipse.jetty.servlet.ServletContextHandler.callContextInitialized(ServletContextHandler.java:434)                                                                                                                     
        at org.eclipse.jetty.server.handler.ContextHandler.startContext(ContextHandler.java:780)                                                                                                                                      
        at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:284)                                                                                                                               
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1322)                                                                                                                                               
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:732)                                                                                                                                           
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:490)                                                                                                                                                     
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)                                                                                                                                        
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:118)                                                                                                                                     
        at org.eclipse.jetty.server.Server.start(Server.java:342)                                                                                                                                                                     
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:100)                                                                                                                                   
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:60)  
        at org.eclipse.jetty.server.Server.doStart(Server.java:290)                           
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)
        at com.dtolabs.rundeck.RunServer.run(RunServer.java:119)                              
        at com.dtolabs.rundeck.RunServer.main(RunServer.java:79)                              
Caused by:                                                                                    
org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalArgumentException: timeNanos out of range 86400754000000"; SQL statement:|select this_.id as y0_ from execution this_ where this_.date_started is not null and this_.date_completed is null and this_.server_nodeuuid is null and this_.date_started<? and (this_.status is null or this_.status<>?) [50000-193]
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)                                     
        at org.h2.message.DbException.get(DbException.java:168)                                                     
        at org.h2.message.DbException.convert(DbException.java:295)                                                 
        at org.h2.command.Command.executeQuery(Command.java:213)                                                    
        at org.h2.jdbc.JdbcPreparedStatement.executeQuery(JdbcPreparedStatement.java:110)                           
        at org.grails.datastore.gorm.GormStaticApi$_withCriteria_closure11.doCall(GormStaticApi.groovy:305)         
        at org.grails.datastore.mapping.core.DatastoreUtils.execute(DatastoreUtils.java:302)                        
        at org.grails.datastore.gorm.AbstractDatastoreApi.execute(AbstractDatastoreApi.groovy:37)                   
        at org.grails.datastore.gorm.GormStaticApi.withCriteria(GormStaticApi.groovy:304)                           
        at rundeck.services.ExecutionService.cleanupRunningJobsAsync(ExecutionService.groovy:652)                   
        at rundeck.services.ExecutionService.cleanupRunningJobsAsync(ExecutionService.groovy:651)                   
        at BootStrap$_closure1.doCall(BootStrap.groovy:377)                                                         
        at grails.util.Environment.evaluateEnvironmentSpecificBlock(Environment.java:327)                           
        at grails.util.Environment.executeForEnvironment(Environment.java:320)                                      
        at grails.util.Environment.executeForCurrentEnvironment(Environment.java:296)                               
        at org.eclipse.jetty.server.handler.ContextHandler.callContextInitialized(ContextHandler.java:788)          
        at org.eclipse.jetty.servlet.ServletContextHandler.callContextInitialized(ServletContextHandler.java:434)   
        at org.eclipse.jetty.server.handler.ContextHandler.startContext(ContextHandler.java:780)                    
        at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:284)             
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1322)                             
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:732)                         
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:490)                                   
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)                      
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:118)                   
        at org.eclipse.jetty.server.Server.start(Server.java:342)                                                   
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:100)                 
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:60)                        
        at org.eclipse.jetty.server.Server.doStart(Server.java:290)                                                 
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)                      
        at com.dtolabs.rundeck.RunServer.run(RunServer.java:119)                                                    
        at com.dtolabs.rundeck.RunServer.main(RunServer.java:79)                                                    
Caused by:                                                                                                          
java.lang.IllegalArgumentException: timeNanos out of range 86400754000000                                           
        at org.h2.value.ValueTimestamp.<init>(ValueTimestamp.java:54)                                               
        at org.h2.value.ValueTimestamp.fromDateValueAndNanos(ValueTimestamp.java:68)                                
        at org.h2.value.ValueTimestamp.fromMillisNanos(ValueTimestamp.java:95)                                      
        at org.h2.store.Data.readValue(Data.java:779)                                                               
        at org.h2.index.PageDataLeaf.readRow(PageDataLeaf.java:624)                                                 
        at org.h2.index.PageDataLeaf.getRowAt(PageDataLeaf.java:334)                                                
        at org.h2.index.PageDataCursor.nextRow(PageDataCursor.java:101)                                             
        at org.h2.index.PageDataCursor.next(PageDataCursor.java:67)                                                 
        at org.h2.index.IndexCursor.next(IndexCursor.java:305)                                                      
        at org.h2.table.TableFilter.next(TableFilter.java:498)                                                      
        at org.h2.command.dml.Select.queryFlat(Select.java:542)                                                     
        at org.h2.command.dml.Select.queryWithoutCache(Select.java:655)                                             
        at org.h2.command.dml.Query.query(Query.java:341)                                                           
        at org.h2.command.dml.Query.query(Query.java:309)                                                           
        at org.h2.command.dml.Query.query(Query.java:36)                                                            
        at org.h2.command.CommandContainer.query(CommandContainer.java:110)                                         
        at org.h2.command.Command.executeQuery(Command.java:201)                                                    
        at org.h2.jdbc.JdbcPreparedStatement.executeQuery(JdbcPreparedStatement.java:110)                           
        at org.grails.datastore.gorm.GormStaticApi$_withCriteria_closure11.doCall(GormStaticApi.groovy:305)         
        at org.grails.datastore.mapping.core.DatastoreUtils.execute(DatastoreUtils.java:302)                        
        at org.grails.datastore.gorm.AbstractDatastoreApi.execute(AbstractDatastoreApi.groovy:37)                   
        at org.grails.datastore.gorm.GormStaticApi.withCriteria(GormStaticApi.groovy:304)                           
        at rundeck.services.ExecutionService.cleanupRunningJobsAsync(ExecutionService.groovy:652)                   
        at rundeck.services.ExecutionService.cleanupRunningJobsAsync(ExecutionService.groovy:651)                   
        at BootStrap$_closure1.doCall(BootStrap.groovy:377)                                                         
        at grails.util.Environment.evaluateEnvironmentSpecificBlock(Environment.java:327)                           
        at grails.util.Environment.executeForEnvironment(Environment.java:320)                                      
        at grails.util.Environment.executeForCurrentEnvironment(Environment.java:296)                               
        at org.eclipse.jetty.server.handler.ContextHandler.callContextInitialized(ContextHandler.java:788)          
        at org.eclipse.jetty.servlet.ServletContextHandler.callContextInitialized(ServletContextHandler.java:434)   
        at org.eclipse.jetty.server.handler.ContextHandler.startContext(ContextHandler.java:780)                    
        at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:284)             
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1322)                             
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:732)                         
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:490)                                   
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)                      
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:118)                   
        at org.eclipse.jetty.server.Server.start(Server.java:342)                                                   
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:100)                 
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:60)                        
        at org.eclipse.jetty.server.Server.doStart(Server.java:290)                                                 
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:69)                      
        at com.dtolabs.rundeck.RunServer.run(RunServer.java:119)                                                    
        at com.dtolabs.rundeck.RunServer.main(RunServer.java:79)                                                    
2017-03-20 21:38:53.373:INFO:oejs.ServerConnector:main: Started ServerConnector@59e33278{SSL-HTTP/1.1}{0.0.0.0:4443}
```

**How to reproduce Behavior**
- Fresh install of 2.6.9 on Ubuntu 16.04
- Update to 2.7.3

I'm using Puppet w/ my fork of the Rundeck module (see the PR here:  https://github.com/voxpupuli/puppet-rundeck/pull/316).

**Questions**
- Did I miss something from [the upgrade guide??](http://rundeck.org/docs/upgrading/index.html)

**Comment by dandunckelman** [comment]
I tried a restoring data from 2.6.9 to a fresh install w/ v2.7.3 and got the same error.

**Comment by pajel** [comment]
Same issue here.

**Comment by dandunckelman** [comment]
@gschueler 
- Did I miss something from [the upgrade guide](http://rundeck.org/docs/upgrading/index.html)?
- Any tests I can do to help?
- Any more info I can provide to help?

**Comment by gschueler** [comment]
@dandunckelman did you try including the `;mv_store=false` in the dataSource config?

the only reference i can find to that error "timeNanos out of range" implies that the newer version of h2 used in 2.7 is detecting a corrupt value in the DB.  https://groups.google.com/forum/#!searchin/h2-database/timeNanos$20out$20of$20range%7Csort:relevance/h2-database/X3JDJIBjeHc/wKuM0qQfDAAJ

**Comment by dandunckelman** [comment]
@gschueler I've tried replicating my production VM locally by doing the following:
- Spin up VM and install v2.6.9
- Restore prod data to new VM
- Update `dataSource` config to add `;mv_store=false`
- Install v2.7.3
- Restart the service

And I get the same error.

Would you do a different process?

I'm in the midst of trying this process:
- Spin up VM and install v2.7.3 with the `dataSource` set with `;mv_store=false`
- Restore prod data
- Restart service

**Comment by gschueler** [comment]
how are you doing "restore prod data"?

**Comment by dandunckelman** [comment]
@gschueler I still get the same error on the 2nd process I outlined.

Please let me know if I should try something else.

Since you mentioned a potentially corrupt value in the DB, I suppose I could hunt for those bad values and fix them?  I'm not sure how, but I'll have to revisit that.

**Comment by dandunckelman** [comment]
@gschueler I'm following the process outlined here:  http://rundeck.org/docs/administration/backup-and-recovery.html

**Comment by dandunckelman** [comment]
@gschueler I'm not sure how I can access the H2 database.  I assume I'd be able to do something like what is described [here](http://h2database.com/html/tutorial.html#tutorial_starting_h2_console)

But IDK where the h2 jar lives.

**Comment by gschueler** [comment]
@dandunckelman h2 jar would be in `server/exp/webapp/WEB-INF/lib/h2-1.4.193.jar` or equivalent path under `/var/lib/rundeck`

if using the original h2 database files is causing this problem, you might have to do the project export/import sequence, starting from your 2.6.9 install.

**Comment by dandunckelman** [comment]
@gschueler thanks.  I forgot I had a `find` running, which showed the same.

Anyway, I was able to start the h2 console with this:

    cd /var/lib/rundeck/exp/webapp/WEB-INF/lib/
    java -jar h2-1.3.176.jar -webAllowOthers

Then opened a web browser to http://MACHINE_DNS_OR_IP:8082, which showed this:

![image](https://cloud.githubusercontent.com/assets/1153702/24930771/98b96d74-1ebf-11e7-953b-438f2c2d7d6b.png)
- **NOTE:** I updated the `JDBC URL` per the value I had set in `/etc/rundeck/rundeck-config.groovy`

Then clicking `Connect`, and I see this:

![image](https://cloud.githubusercontent.com/assets/1153702/24930890/fc7cbf82-1ebf-11e7-830c-47012b40fe23.png)

Then I did queries like this on each table to see if any column has the value "86400754000000":

    select * from WORKFLOW_WORKFLOW_STEP
        where 86400754000000 in (
            select column_name from INFORMATION_SCHEMA.columns
                where table_name = 'WORKFLOW_WORKFLOW_STEP'
        )
    ;

But got no matches.  Should I do different queries?

> if using the original h2 database files is causing this problem, you might have to do the project export/import sequence, starting from your 2.6.9 install.

Is that different from [this](http://rundeck.org/docs/administration/backup-and-recovery.html)?

I'm almost to the point of either trashing all the data and starting over or trying [to migrate to MySQL](https://github.com/rundeck/rundeck/blob/v2.8.0/docs/en/administration/07-relational-database.md)

**Comment by gschueler** [comment]
@dandunckelman the column value causing the problem is i think a Date column, the stacktrace indicates that.  You might try the h2 `Recover` tool, see http://h2database.com/html/tutorial.html#command_line_tools

that [page](http://rundeck.org/docs/administration/backup-and-recovery.html) has two methods: backup data/logs dirs, and then recover in place, and using the project export/import. I was under the impression you were using the h2 1.3 data files, and starting up rundeck 2.7 with those (which uses h2 1.4).  Instead, you could start with a clean install of rundeck 2.7, creating new h2 data files, and import project archive files into the clean database.

**Comment by gschueler** [comment]
@dandunckelman i would recommend migrating to mysql instead of h2

**Comment by dandunckelman** [comment]
@gschueler thanks for the info.

I'm currently trying the project export/import process.

Hopefully, I'll have good news later today.

**Comment by dandunckelman** [comment]
@gschueler I just tested importing the exports I made on a local VM, and they worked as expected on v2.8.2!

So, I think I'll do this in production:
- Export the projects again, so I have them up to date
- Delete VM
- Spin up new VM installing v2.8.2
- Import archives
- Celebrate

Thanks for your help.  I'll close this as I think a fix for my issue cannot be done by Rundeck itself (and its update process).
